### PR TITLE
Some general improvements on the k8s documentation

### PR DESCRIPTION
**Pull request by iuriaranda**
These are just some general improvements and fixes of our K8s documentation:

- some lint error fixes
- update monitoring documentation to fit the current offering
- update the k8s walk-through document to include some relevant topics (monitoring & updates)

### CODE REVIEW

File: kubernetes/monitoring.md [hunk 1]
```diff
@@ -9,13 +9,18 @@ As of this writing there isn't an operator setup yet for Grafana, but you can ad
...
+- All `ServiceMonitors` and `PrometheusRules` you define need to have the `prometheus` label (any value will do).
+- The namespace where you create your `ServiceMonitors` and `PrometheusRules` need to have the `prometheus` label too (any value will do).
+
+   ```bash
+   kubectl label namespace yournamespace prometheus=true
+   ```
 
 Custom Grafana Dashboards can also be added to this same namespace by creting a `ConfigMap` with a `grafana_dashboard` label (any value will do), containing the dashboard's json data. Please make sure that `"datasources"` are set to `"Prometheus"` in your dashboards!
 
-*Note: even if these objects are created in the `application-monitoring` namespace, Prometheus can scrape metric targets in all namespaces.*
```
Comment (duboisph) [inline kubernetes/monitoring.md:18]: Sidenote: I'd also like to remove this namespace from our stack's install. However looks like some customers do use it.

File: kubernetes/monitoring.md [hunk 2]
```diff
@@ -86,3 +99,9 @@ data:
   grafana-dashboard-myapplication.json: |
 { <Grafana dashboard json> }
 ```
+
+You can get inspired by some of the dashboards already deployed in the cluster:
+
+```bash
+kubectl get configmaps -l grafana_dashboard=grafana_dashboard=cluster-monitoring --all-namespaces
```
Comment (duboisph) [inline kubernetes/monitoring.md:106]: Typo, duplicate key `grafana_dashboard`.

File: kubernetes/walk-through.md [hunk 3]
```diff
@@ -81,15 +85,14 @@ With Helm, you create self contained packages for a specific piece of a deployme
...
 By using a Template engine, a [Go function library](http://masterminds.github.io/sprig/) and the use of a `Values.yaml` file, you can build a template of a specific piece and re-use that for multiple deployments.
 
-The [Helm documentation](https://docs.helm.sh/) is quite good, but this section highlights 2 specific features that should be used from the start to create good reusable Charts:
+It's worth noting that [Tiller won't be deployed anymore on K8s clusters](https://changelog.skyscrapers.eu/kubernetes/2020/04/10/helm3.html), which means that you should be using Helm v3.
 
-* [Dependencies](https://docs.helm.sh/chart-best-practices/#requirements)
-* [Lifecycle Hooks](https://docs.helm.sh/developing-charts/#chart-lifecycle-hooks)
+The [Helm documentation](https://helm.sh/docs/) is quite good and explanatory, and the [best practices section](https://helm.sh/docs/chart_best_practices/) highlights some of the important topics around chart development.
```
Comment (duboisph) [inline kubernetes/walk-through.md:90]: s/highlights/highlight

File: kubernetes/walk-through.md [hunk 4]
```diff
@@ -308,19 +311,48 @@ You can find some examples in the `kube2iam` official documentation: <https://gi
...
+
+We have added a Prometheus alert to catch this taint and you can [follow the actions described in the runbook](https://github.com/skyscrapers/documentation/tree/master/runbook.md#alert-name-nodewithimpairedvolumes) when this happens.
+
+## Monitoring
+
+Cluster and application monitoring is a quite extensive topic by itself, so there's a specific document for it [here](./monitoring.md).
+
+## Cluster updates & rollouts
+
+As part of our responsabilities, we continuously roll improvements (upgrades, updates, bug fixes and new features). Depending on the type of improvement, the impact on platform usage and application varies anywhere between nothing to having (a small) downtime. Below an overview for the most common types of improvements. More exceptional types will be handled separately.
```
Comment (duboisph) [inline kubernetes/walk-through.md:332]: s/responsabilities/responsibilities/

File: kubernetes/walk-through.md (hunk 3 above)
Comment (iuriaranda) [reply kubernetes/walk-through.md:90]: 🤔 in this case the subject is "best practices **section**", which is third-person singular. So it should be "highlights", no?

**Pull request merged**